dev/io.openliberty.security.jakartasec.3.0.internal_fat.config/fat/src/io/openliberty/security/jakartasec/fat/config/tests/ConfigurationUserInfoTests.java

```diff
@@ -579,11 +840,276 @@ public void ConfigurationUserInfoTests_invalidUserinfoEndpoint() throws Exceptio
+
+        Page response = runGoodEndToEndTest("UserinfoTestServletRS512", app);
+
+        extraUserinfoSigMismatchValidation(response);
+
+    }
+
+    // TODO - enabled once the userinfo app can handle signing with ES algs
+//    @Test
+//    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+//    public void ConfigurationUserInfoTests_invalidES256JWTUserinfoResponse() throws Exception {
+//
+//        Page response = runGoodEndToEndTest("UserinfoTestServletES256", app);
+//
+//        extraUserinfoSigMismatchValidation(response);
+//
+//    }
+//
+//    @Test
+//    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+//    public void ConfigurationUserInfoTests_invalidES384JWTUserinfoResponse() throws Exception {
+//
+//        Page response = runGoodEndToEndTest("UserinfoTestServletES384", app);
+//
+//        extraUserinfoSigMismatchValidation(response);
+//
+//    }
+//
+//    @Test
+//    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+//    public void ConfigurationUserInfoTests_invalidES512JWTUserinfoResponse() throws Exception {
+//
+//        Page response = runGoodEndToEndTest("UserinfoTestServletES512", app);
+//
+//        extraUserinfoSigMismatchValidation(response);
+//
+//    }
+
+    /**
+     * Tset that with the userinfo endpoint returning a jwt signed with HS256 instead of RS256 we'll still be able to access the app because we don't need the information from
+     * userinfo
+     *
+     * @throws Exception
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+    public void ConfigurationUserInfoTests_invalidHS256JWTUserinfoResponse() throws Exception {
+
+        Page response = runGoodEndToEndTest("UserinfoTestServletHS256", app);
+
+        extraUserinfoSigMismatchValidation(response);
+
+    }
+
+    /**
+     * Tset that with the userinfo endpoint returning a jwt signed with HS384 instead of RS256 we'll still be able to access the app because we don't need the information from
+     * userinfo
+     *
+     * @throws Exception
+     */
+    @Test
+    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+    public void ConfigurationUserInfoTests_invalidHS384JWTUserinfoResponse() throws Exception {
+
+        Page response = runGoodEndToEndTest("UserinfoTestServletHS384", app);
+
+        extraUserinfoSigMismatchValidation(response);
+
+    }
+
+    /**
+     * Tset that with the userinfo endpoint returning a jwt signed with HS512 instead of RS256 we'll still be able to access the app because we don't need the information from
+     * userinfo
+     *
+     * @throws Exception
+     */
+    @Test
+    @ConditionalIgnoreRule.ConditionalIgnore(condition = skipIfJsonUserInfo.class)
+    public void ConfigurationUserInfoTests_invalidHS512JWTUserinfoResponse() throws Exception {
```

Comment:
I don't think we need algorithm mismatch tests for RS256 + RS384|RS512|HS256|HS384|HS512 because they're all testing the same thing. A max of two should be plenty sufficient: RS256 + a mismatched RS algorithm, and RS256 + some non-RS algorithm.

Reply:
reduced to just testing RS256 with RS512 and HS384.